src/libraries/Common/src/System/IO/ChunkedMemoryStream.cs

```diff
@@ -43,7 +50,9 @@ public override void Write(byte[] buffer, int offset, int count)
                     if (remaining > 0)
                     {
                         int toCopy = Math.Min(remaining, count);
-                        Buffer.BlockCopy(buffer, offset, _currentChunk._buffer, _currentChunk._freeOffset, toCopy);
+                        ReadOnlySpan<byte> source = buffer.Slice(offset, toCopy);
+                        Span<byte> destination = new Span<byte>(_currentChunk._buffer, _currentChunk._freeOffset, toCopy);
+                        source.CopyTo(destination);
```

Comment:
offset and count aren't needed any more, so this can be simplified a bit, e.g. something like:
```C#
while (!buffer.IsEmpty)
{
    if (_currentChunk is not null)
    {
        int remaining = _currentChunk._buffer.Length - _currentChunk._freeOffset;
        if (remaining > 0)
        {
            int toCopy = Math.Min(remaining, buffer.Length);

            buffer.Slice(0, toCopy).CopyTo(_currentChunk._buffer.AsSpan(_currentChunk._freeOffset));

            buffer = buffer.Slice(toCopy);
            _totalLength += toCopy;
            _currentChunk._freeOffset += toCopy;
            continue;
        }
    }

    AppendChunk(buffer.Length);
}
```

Reply:
I've made this change. I was unable to replace the span ctor with `AsSpan` due to the required assembly not being referenced by that project.

Reply:
Which project?

Reply:
System.Net.WebClient

Reply:
You can add System.Memory at https://github.com/dotnet/runtime/blob/665c65a8929a866b79409b832f0fa6af14cbf230/src/libraries/System.Net.WebClient/src/System.Net.WebClient.csproj#L32 if you'd like,  but using the ctor is also fine.